Lenovo enos bugfixes

amuraleedhar opened the pull request:
SUMMARY

Creating the config module for enos from Lenovo.

ISSUE TYPE

- Bugfix Pull Request

COMPONENT NAME

lib/ansible/modules/network/enos/enos_config.py
lib/ansible/module_utils/network/enos/enos.py

ANSIBLE VERSION

ansible 2.5.0
config file = /etc/ansible/ansible.cfg
configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible-2.5.0-py2.7.egg/ansible
executable location = /usr/local/bin/ansible
python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]

ADDITIONAL INFORMATION

This is an effort to create config modules for enos based switches from Lenovo.

Commit:
Bug Fixes to issues foubd during testing

Commit:
Removing blank line

Commit:
undefined-variable Undefined variable 'run_commands' and training new line

gundalow commented on lib/ansible/modules/network/enos/enos_config.py:
```diff
@@ -248,32 +243,35 @@ def main():
 
         match=dict(default='line', choices=['line', 'strict', 'exact', 'none']),
-        replace=dict(default='line', choices=['line', 'block', 'config']),
+        replace=dict(default='line', choices=['line', 'block']),
```
`DOCUMENTATION:` for replace needs updating

gundalow commented on lib/ansible/modules/network/enos/enos_config.py:
```diff
@@ -248,32 +243,35 @@ def main():
+        replace=dict(default='line', choices=['line', 'block']),
 
         config=dict(),
+        defaults=dict(type='bool', default=False),
```
`DOCUMENTATION:` for `defaults:` needs adding

Remember you can use `type: bool` rather than `choices:` in DOCUMENTATION.

amuraleedhar replied:
Reverting the changes back.

gundalow commented on lib/ansible/modules/network/enos/enos_config.py:
```diff
@@ -248,32 +243,35 @@ def main():
 
         config=dict(),
+        defaults=dict(type='bool', default=False),
+        passwords=dict(type='bool', default=False),
```
`DOCUMENTATION:` for `passwords:` needs adding.
From the name `passwords:` it isn't clear to me what this is doing. Is there a better name?

amuraleedhar replied:
Reverting the changes back.

amuraleedhar replied:
Reverting the changes back.

gundalow commented on lib/ansible/modules/network/enos/enos_config.py:
```diff
@@ -248,32 +243,35 @@ def main():
         backup=dict(type='bool', default=False),
-        comment=dict(default=DEFAULT_COMMIT_COMMENT),
-        admin=dict(type='bool', default=False)
+        save=dict(type='bool', default=False),
```
`DOCUMENTATION:` for `save:` needs adding.

amuraleedhar replied:
Reverting the changes back.

gundalow reviewed (changes requested):
Unit tests are failing

https://app.shippable.com/github/ansible/ansible/runs/47205/2/tests

`test/units/modules/network/enos/test_enos_config.py`

Commit:
Reverting and Modifying changes with both Unit test and Integrated test with Devices passing

amuraleedhar commented:
This Pull Reqest is done to attend the integrated test failure with real ENOS devices. The IT tests where failing at candidate.load(module.params['src']) at get_candidate method of enos_config.py. I changed it into candidate.loadfp(module.params['src']) to pass it on real devices. Then The unit tests began failing. So I added this logic in module and now both are passing
 try:
            candidate.loadfp(module.params['src'])
        except IOError:
            candidate.load(module.params['src'])

amuraleedhar commented:
Please merge this Bug Fix PR. Expansion of features will be done in another PR

gundalow commented:
Merged into `devel`, will be released in `2.5.0`
